runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java

```diff
@@ -202,12 +274,38 @@ public void onGameTick(GameTick event)
 				int hit = specialWeapon == TONALZTICS_OF_RALOS ? specialAttackHits : getHit(specialWeapon, lastSpecHitsplat);
 				specialAttackHit(specialWeapon, hit, lastSpecTarget);
 			}
+			else if (specialWeapon == SpecialWeapon.BANDOS_GODSWORD && TEKTON_VARIANTS.contains(lastSpecTarget.getId()))
+			{
+				specialAttackHit(specialWeapon, 10, lastSpecTarget);
+			}
 
 			specialWeapon = null;
 			lastSpecHitsplat = null;
 			secondToLastSpecHitsplat = null;
 			lastSpecTarget = null;
 			specialAttackHits = 0;
+			specialAttempted = false;
+			hitpointsChanged = false;
+
+		}
+		else if (specialAttempted)
+		{
+			if (hitpointsChanged)
```

Comment:
You're right, but to my knowledge the only way this could occur is if you 1-ticked both actions of a lamp->special attack. I don't believe unintentional hitpoints xp that you can receive same tick currently exists

Reply:
?
![image](https://github.com/runelite/runelite/assets/309739/3c819ce7-03cb-47a4-a836-3ef856a6261b)
Both the false assignments are behind specialWeapon != null && lastSpecTarget != null? Am I missing something?

Reply:
Oh, I think I misunderstood your point, yup there is a missing check when that variable is assigned to true. it needs to return if special weapon is null

Reply:
> Oh, I think I misunderstood your point, yup there is a missing check when that variable is assigned to true. it needs to return if special weapon is null

Although, that variable is behind a invokelater so I'll have to think about that one